controllers/etcd_controller.go

```diff
@@ -791,142 +797,129 @@ func (r *EtcdReconciler) getStatefulSetFromEtcd(etcd *druidv1alpha1.Etcd, values
 	return decoded, nil
 }
 
-func (r *EtcdReconciler) reconcileCronJob(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd, values map[string]interface{}) (*batchv1.CronJob, error) {
-	logger.Info("Reconcile etcd compaction cronjob")
-
-	var cronJob batchv1.CronJob
-	err := r.Get(ctx, types.NamespacedName{Name: getCronJobName(etcd), Namespace: etcd.Namespace}, &cronJob)
-
-	//If backupCompactionSchedule is present in the etcd spec, continue with reconciliation of cronjob
-	//If backupCompactionSchedule is not present in the etcd spec, do not proceed with cronjob
-	//reconciliation. Furthermore, delete any already existing cronjobs corresponding with this etcd
-	backupCompactionScheduleFound := false
-	if values["backup"].(map[string]interface{})["backupCompactionSchedule"] != nil {
-		backupCompactionScheduleFound = true
+func (r *EtcdReconciler) reconcileServiceAccount(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd, values map[string]interface{}) error {
+	logger.Info("Reconciling serviceaccount")
+	var err error
+	decoded := &v1.ServiceAccount{}
+	serviceAccountPath := getChartPathForServiceAccount()
+	chartPath := getChartPath()
+	renderedChart, err := r.chartApplier.Render(chartPath, etcd.Name, etcd.Namespace, values)
+	if err != nil {
+		return err
 	}
-
-	if !backupCompactionScheduleFound {
-		if err == nil {
-			err = r.Delete(ctx, &cronJob, client.PropagationPolicy(metav1.DeletePropagationForeground))
```

Comment:
I don't think that we can simply remove the deletion of cron jobs because people might have already enabled that feature, so these objects will remain in the cluster after switching to `Job` in https://github.com/gardener/etcd-druid/pull/235.
In general I have to admit, I don't like that this PR tries to include both: Adding new resource for Backup-Restore and removing the `CronJob` behavior. Why?
- It makes it depended on https://github.com/gardener/etcd-druid/pull/235.
- It breaks the main branch after merging.
- It makes reviews harder to understand.
- It makes the change history complex.
Not sure why that decision was made https://github.com/gardener/etcd-druid/pull/233#pullrequestreview-766861481 (/cc @shreyas-s-rao).

Reply:
Yes, I already took care of cron job related codes in my PR. Would add a release note. Thanks.